spec/lib/msf/core/rpc/v10/job_state_tracker_spec.rb

```diff
@@ -0,0 +1,159 @@
+require 'spec_helper'
+
+require 'msf/core/rpc/v10/job_state_tracker'
+
+RSpec.describe JobStateTracker do
+
+  context "With a 0.1 second ttl " do
+    before(:each) do
+      @job_state_tracker = JobStateTracker.new(result_ttl=0.1)
+    end
+
+    context "A job has completed" do
+      before(:each) do
+        @job_id = "super_random_job_id"
+        @generic_result = "This is a job that has finished"
+        @job_state_tracker.completed(@job_id, @generic_result)
+      end
+
+      it 'should show as finished' do
+        expect(@job_state_tracker.finished? @job_id).to eq(true)
+      end
+
+      context "we wait out the ttl" do
+        before(:each) do
+          sleep 0.1
+        end
+
+        it 'should not show as finished' do
+          expect(@job_state_tracker.finished? @job_id).to eq(false)
+        end
+      end
+    end
+  end
+
+  context "With default options" do
```

Comment:
What are your thoughts on using rspec's `let` syntax here:

```ruby
context "with default options" do
    let(:job_state_tracker) { described_class.new }
    let(:job_id) { "1234" }

    describe "#waiting" do
       before(:each) do
          job_state_tracker.waiting(job_id)
       end
      it { expect(job_state_tracker).to have_waiting job_id }
      it { expect(job_state_tracker).not_to have_running job_id }    
      it { expect(job_state_tracker).not_to have_finished job_id }    
    end
end
```

Extra notes:
- The `#waiting` notation comes from http://www.betterspecs.org/#describe
- The `described_class` comes from the argument given to `Rspec.describe` on line 5 - https://relishapp.com/rspec/rspec-core/docs/metadata/described-class
- I haven't tested any of this, your mileage may vary

Reply:
I'll poke about at it, it certainly looks nicer